### PR TITLE
fix test failures caused by newer morecantile version and pre-commit warnings

### PR DESCRIPTION
**Pull request by philvarner**
1. version 2.1.4 added additional tilesetmatrix options, which causes several tests to fail. Fixed those tests by updating the expected counts.
2. the github pre-commit hooks were failing because of numerous minor issues, like unused imports, f-strings without replacements, etc. Fixed all of these so the pre-commit hooks run cleanly now.

next step will be to actually update from upstream

### CODE REVIEW

File: titiler/application/tests/routes/test_tms.py [hunk 1]
```diff
@@ -6,7 +6,7 @@ def test_tilematrix(app):
     response = app.get("/tileMatrixSets")
     assert response.status_code == 200
     body = response.json()
-    assert len(body["tileMatrixSets"]) == 12  # morecantile has 10 defaults
+    assert len(body["tileMatrixSets"]) == 13  # morecantile has 13 defaults
```
Comment (philvarner) [inline titiler/application/tests/routes/test_tms.py:9]: morecantile update added more

File: titiler/mosaic/tests/test_factory.py [hunk 2]
```diff
@@ -128,8 +129,9 @@ def test_MosaicTilerFactory():
 
 root = "http://testserver/mosaic"
 
-def get_link_by_rel(response_body: dict, rel: str) -> str:
-    return next((x["href"] for x in response_body["links"] if x["rel"] == rel), None)
+
+def _get_link_by_rel(response_body: dict, rel: str) -> str:
+    return next((x["href"] for x in response_body["links"] if x["rel"] == rel), "")
```
Comment (philvarner) [inline titiler/mosaic/tests/test_factory.py:134]: for some reason, None is allowed in python 3.9 and not allowed in python 3.8 🤷🏻

File: titiler/mosaic/tests/test_factory.py [hunk 3]
```diff
@@ -143,36 +145,35 @@ def test_mosaics_basic(set_env, client):
...
+    )
     assert r.status_code == 201
     mosaic_id = r.json()["id"]
     assert mosaic_id
 
     location_header_url = r.headers["location"]
     assert location_header_url == f"{root}/mosaics/{mosaic_id}"
 
-    assert location_header_url == get_link_by_rel(r.json(), "self")
+    assert location_header_url == _get_link_by_rel(r.json(), "self")
```
Comment (philvarner) [inline titiler/mosaic/tests/test_factory.py:158]: making this private after it threw a warning

File: titiler/mosaic/titiler/mosaic/resources/models.py [hunk 4]
```diff
@@ -1,12 +1,11 @@
 """Titiler.mosaic Models."""
 
-from cogeo_mosaic.mosaic import MosaicJSON
```
Comment (philvarner) [inline titiler/mosaic/titiler/mosaic/resources/models.py:3]: unused

**Review by abarciauskas-bgse — APPROVED** [review]

**Pull request merged**